### ISSUE TITLE
Jupyter kernel restarts when the `ipynb` file is renamed

### ISSUE THREAD

**Issue by apcamargo** [microsoft/vscode-jupyter#11010]
### Applies To

- [X] Notebooks (.ipynb files)
- [ ] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

When I rename a notebook I'm working in (by right-clicking the file in the left pane) the Jupyter server restarts. In JupyterLab you can rename a `ipynb` file while working on it and I expected the same behaviour in VSCode.

### VS Code Version

1.69.2

### Jupyter Extension Version

v2022.6.1201981810

### Coding Language and Runtime Version

Python 3.9.2

### Running Jupyter locally or remotely?

Local

**Comment by rchiodo** [comment]
Thanks for the bug.

Do you mean the kernel is restarted? 

That would make sense the way VS code is structured. The kernel is tied to the document, renaming the document destroys the old one, hence killing the kernel.

We should probably (at least in the rename case), not kill the old kernel.

This is related to #1378

**Comment by apcamargo** [comment]
Yes, the kernel restarts.

I really like the way VSCode ties kernels to files, since this makes my workflow much easier (I don't have to start a server to to write a quick notebook that I'll delete soon afterwards). But the renaming scenario should be handled in a better way. A warning message would probably suffice.

**Comment by DonJayamanne** [comment]
The problem here is, the active kernel is not switched from one notebook to the other, hence the old kernel id destroyed and a new one started for the newly renamed notebook.

**Comment by DonJayamanne** [comment]
Duplicate of https://github.com/microsoft/vscode-jupyter/issues/8215

**Comment by DonJayamanne** [comment]
Related issue https://github.com/microsoft/vscode-jupyter/issues/9517 as we need a way to connect a notebook to an active notebook kernel.

**Comment by DonJayamanne** [comment]
@roblourens @rebornix  do you think we can have some kind of event that would help Jupyter extension determine whether the user renamed a notebook from `x.ipynb` to `y.ipynb`, once we have that we can try to keep the same instance of the kernel for the user.

**Comment by roblourens** [comment]
Can you use `vscode.workspace.onDidRenameFiles` for this?

**Comment by IanMatthewHuff** [comment]
Closing again as I agree with the duplication of #8215. I'll transfer over the info in the comments.

**Comment by DonJayamanne** [comment]
@IanMatthewHuff I'd like to keep this open and sepearate from #8215 
They are different.
* With #8215, we want to ensure the same kernel is selected, 
    * E.g. assume you have kernel A selected with notebook X.ipynb.
    * Now the notebook is closed, and you rename the file.
    * We need to ensure when you open the renamed file the kernel points to `A`.
    * Similar even if the notebook is opened (but user has not executed any cells) 
* This issue pertains to keeping the active kernel session insteaad of disposing it
    * That requires a bit more work, its not the same as selecting the right controller.
    * We need to ensure we select the same active kernel session, not just the controler, as selecting the controller will not address this issue, as the kernel state would be lost.
    * & this requires work to be done in the kernel layer to ensure we can transfer the kernel from one resource to another. Today a kernel and a notebook have a 1-1 relationship and changing that requires a bit of work, that can be done completely independant of the first issue (i.e. independant of issue#8215)

I've assigned this to you and me, as we'll need to work on this together. As it touches both kernel picker as well as kernel state/execution layers.
THis PR https://github.com/microsoft/vscode-jupyter/pull/10573 will be required to resolve the problem described in this issue

**Comment by DonJayamanne** [comment]
Closing this issue for now as this hasn't been reported anyone else for over a year, 
Closing as out of scope for now.

**Comment by apcamargo** [comment]
Is this issue truly fixed? I still lose my section when the file is renamed.

**Comment by Matt-BF** [comment]
Hey, I have been having the same issue and just found this thread. Usually it is a minor problem, but sometimes when I decide to turn a quick and dirty notebook into something more permanent (thus renaming it) and when having cells with longer computations, the workflow gets broken and re-running everything can be a bit of a waste of computational resources

**Comment by DonJayamanne** [comment]
Use `onWillRenameFiles` event to migrate the kernels.
Blocked on https://github.com/microsoft/vscode-jupyter/issues/9517
Even if we migrate the kernels, we need to create a controller thats associated with a live kernel. and related to #9517

**Comment by DonJayamanne** [comment]
Closing as not planned
